cmake 3.31 is broken: CMake Error: Could not find CMAKE_ROOT !!!

See this error in the CI, which can be reproduced:

```
CMake Error: Could not find CMAKE_ROOT !!!
CMake has most likely not been installed correctly.
Modules directory not found in
/usr/local/share/cmake-3.31
CMake Error: Error executing cmake::LoadCache(). Aborting.
```

reproduce:
https://gist.github.com/njzjz/1c8f360ebc4457bf0d0f6ea86978b78b

CI log: https://github.com/deepmodeling/deepmd-kit/actions/runs/11758764445/job/32757483294#step:3:1

--- a/src/cmake/__init__.py
+++ b/src/cmake/__init__.py
@@ -49,7 +49,8 @@ def _program(name: str, args: Iterable[str]) -> int:
 def _program_exit(name: str, *args: str) -> NoReturn:
     if sys.platform.startswith("win"):
         raise SystemExit(_program(name, args))
-    os.execl(os.path.join(CMAKE_BIN_DIR, name), name, *args)
+    cmake_exe = os.path.join(CMAKE_BIN_DIR, name)
+    os.execl(cmake_exe, cmake_exe, *args)
 
 
 def ccmake() -> NoReturn:

